nscplugin/src/test/scala/scala/scalanative/NIRCompiler.scala

```diff
@@ -0,0 +1,150 @@
+package scala.scalanative
+
+import java.nio.file.{Files, Path}
+import java.io.{File, PrintWriter}
+import java.net.URLClassLoader
+
+object NIRCompiler {
+
+  private val allow: String => Boolean =
+    n => n.startsWith("scala.scalanative.api.") || !n.startsWith("scala.")
```

Comment:
This change should fix the linkage errors. In general we need to filter out both `scala.*` and `dotty.*` classes
```suggestion
  private val allow: String => Boolean = {
    case s"scala.scalanative.api.${_}" => true
    case s"scala.${_}"                 => false
    case s"dotty.${_}"                 => false
    case _                             => true
  }
```